jupyterlab/handlers/ydoc.py

```diff
@@ -23,6 +27,11 @@ def __init__(self):
     def source(self):
         return str(self._ysource)
 
+    @source.setter
+    def source(self, value):
+        with self._ydoc.begin_transaction() as t:
+            self._ysource.push(t, value)
```

Comment:
We should probably delete all prior characters before pushing, even though it is only initialized once so it should not be a problem.